fms/datasets/text.py

```diff
@@ -0,0 +1,79 @@
+import torch
+from torch.utils.data import Dataset
+import requests
+from fms.utils import tokenizers
+
+
+class FromString(Dataset):
+    """
+    Creates a dataset from a single text string and tokenizer.
+    Since all data comes from a single text, there are no bos/eos tokens used.
+    A pad token if specified, is used only on the final row. i.e.
+    `pad_token=None` is similar to drop_last in DataLoader.
+    """
```

Comment:
We might want to have BOS if we are using this for training any real models. Are we using this in the future for this case or is this just for testing?

Reply:
just for testing -- a large model would always need more data than what fits in a single string in memory. we can add features later but the easiest thing to do is just not specify a pad token (behaves like drop last).
I think a bos token that only appears once in a single row might not be that helpful to training anyway?